src/app/api/imageInput/route.ts

```diff
@@ -0,0 +1,170 @@
+import { ChatOpenAI } from "@langchain/openai";
+import { HumanMessage } from "@langchain/core/messages";
+import { env } from "@/app/env.mjs";
+import { NextResponse } from "next/server";
+import { jsonToLangchain02, saveDroppedImage } from "@/utils/apiHelper";
+import { z } from "zod";
+import { Message } from "ai/react/dist";
+import { auth } from "@clerk/nextjs";
+import { NextApiResponse } from "next";
+import { StreamingTextResponse, LangChainStream } from "ai";
+import { systemPrompt } from "@/utils/prompts";
+
+const dropZoneInputSchema = z.object({
+  imageName: z.string(),
+  imageType: z.string(),
+  imageSize: z.number(),
+  value: z.string(),
+  userId: z.string(),
+  orgId: z.string(),
+  chatId: z.any(),
+  id: z.string(),
+  imageFile: z.any(),
+  messages: z.any(),
+});
+
+export async function POST(request: Request, response: NextApiResponse) {
+  const formData = await request.formData();
+  const zodMessageString: any = formData.get("zodMessage");
+  const file: any = formData.get("file");
+  const zodMessageObject = JSON.parse(zodMessageString);
+
+  const zodMessage = dropZoneInputSchema.parse({
+    imageName: zodMessageObject.data.imageName,
+    imageType: zodMessageObject.data.imageType,
+    imageSize: zodMessageObject.data.imageSize,
+    value: zodMessageObject.data.value,
+    userId: zodMessageObject.data.userId,
+    orgId: zodMessageObject.data.orgId,
+    chatId: zodMessageObject.data.chatId,
+    id: zodMessageObject.data.id,
+    imageFile: file,
+    messages: zodMessageObject.data.message as Message[],
+  });
+
+  // console.log("zodMessage:", zodMessage);
+
+  const {
+    imageName,
+    imageType,
+    imageSize,
+    value,
+    userId,
+    orgId,
+    chatId,
+    id,
+    messages,
+    imageFile,
+  } = zodMessage;
+  const { orgSlug } = await auth();
+
+  // const _message = messages as unknown as Message[];
+  // console.log("_message", _message);
+  // const url = request.url;
+  // const urlArray = url.split("/");
+
+  if (!formData || !formData.has("file")) {
+    return NextResponse.json(
+      { result: "no data received", success: false },
+      { status: 400 },
+    );
+  }
+  const parts = imageFile.name.split(".");
+  const extension = parts[parts.length - 1];
+  // console.log("messages", messages)
+
+  const msg: any = jsonToLangchain02(messages, systemPrompt);
```

Comment:
iinstead of having a seperate function extend the previous one(jsonToLangchain) if things are working